### PR TITLE
Add unit tests for checking the content of all `.json` files used for completion

### PR DESCRIPTION
**Pull request by jlelong**
@tamuratak Based on https://github.com/tamuratak/LaTeX-Workshop/tree/unittest, I have written some tests to check that every entry in the `.json` files used for completion contain the correct keys and only them.

### CODE REVIEW

File: test/utils.ts [hunk 1]
```diff
@@ -178,3 +178,7 @@ export async function getViewerStatus(pdfFilePath: string) {
         }
     }, process.platform === 'win32' ? 600 : undefined)
 }
+
+export function checkDictkeys(keys: string[], expectedKeys: string[], optKeys: string[] = []): boolean {
+    return keys.every(k => expectedKeys.includes(k) || optKeys.includes(k)) && expectedKeys.every(k => keys.includes(k))
+}
```
Comment (tamuratak) [inline test/utils.ts:184]: - `check` is too general. `assertEqualDictKeys` is better.
- Move to `unittest.test.ts`. The function is only for the test.
- We had better call `assert.deepStrictEqual`: https://nodejs.org/api/assert.html#assertdeepstrictequalactual-expected-message

**Review by tamuratak — CHANGES_REQUESTED** [review]

File: test/unittest.test.ts [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+        return
+    })
+
+    runTestWithFixture('fixture001', 'check default environment .json completion file', () => {
+        const extensionRoot = path.resolve(__dirname, '../../')
+        const file = `${extensionRoot}/data/environments.json`
+        const envs = JSON.parse(fs.readFileSync(file, {encoding: 'utf8'})) as {[key: string]: EnvType}
+        assert.ok(Object.keys(envs).length > 0)
+        Object.keys(envs).forEach(name => {
+            assert.ok(checkDictKeys(Object.keys(envs[name]), ['name'] , ['snippet', 'detail']), file + ': ' + JSON.stringify(envs[name]))
```
Comment (tamuratak) [inline test/unittest.test.ts:47]: Please format the line to multiple lines for the readability.

```ts
            assert.ok(
                checkDictKeys(
                    Object.keys(envs[name]),
                    ['name'],
                    ['snippet', 'detail']
                ),
                file + ': ' + JSON.stringify(envs[name])
            )
```

File: test/unittest.test.ts [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+    })
+
+    runTestWithFixture('fixture001', 'check environments from package .json completion files', () => {
+        const extensionRoot = path.resolve(__dirname, '../../')
+        const files = glob.sync('data/packages/*_env.json', {cwd: extensionRoot})
+        files.forEach(file => {
+            const envs = JSON.parse(fs.readFileSync(path.join(extensionRoot, file), {encoding: 'utf8'})) as {[key: string]: EnvType}
+            assert.ok(Object.keys(envs).length > 0)
+            Object.keys(envs).forEach(name => {
+                assert.ok(checkDictKeys(Object.keys(envs[name]), ['name', 'snippet', 'detail', 'package']), file + ': ' + JSON.stringify(envs[name]))
```
Comment (tamuratak) [inline test/unittest.test.ts:59]: Please format the line to multiple lines for the readability.

File: test/unittest.test.ts [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+    })
+
+    runTestWithFixture('fixture001', 'check commands from package .json completion files', () => {
+        const extensionRoot = path.resolve(__dirname, '../../')
+        const files = glob.sync('data/packages/*_cmd.json', {cwd: extensionRoot})
+        files.forEach(file => {
+            const cmds = JSON.parse(fs.readFileSync(path.join(extensionRoot, file), {encoding: 'utf8'})) as {[key: string]: CmdType}
+            assert.ok(Object.keys(cmds).length > 0)
+            Object.keys(cmds).forEach(name => {
+                assert.ok(checkDictKeys(Object.keys(cmds[name]), ['command', 'snippet'] , ['documentation', 'detail', 'postAction', 'package', 'label']), file + ': ' + JSON.stringify(cmds[name]))
```
Comment (tamuratak) [inline test/unittest.test.ts:83]: Please format the line to multiple lines for the readability.

File: test/unittest.test.ts [hunk 5]
```diff
@@ -0,0 +1,90 @@
...
+        return Promise.resolve()
+    })
+
+    runTestWithFixture('fixture001', 'check default commands .json completion file', () => {
+        const extensionRoot = path.resolve(__dirname, '../../')
+        const file = `${extensionRoot}/data/commands.json`
+        const cmds = JSON.parse(fs.readFileSync(file, {encoding: 'utf8'})) as {[key: string]: CmdType}
+        assert.ok(Object.keys(cmds).length > 0)
+        Object.keys(cmds).forEach(name => {
+            assert.ok(checkDictKeys(Object.keys(cmds[name]), ['command'] , ['snippet', 'documentation', 'detail', 'postAction', 'label']), file + ': ' + JSON.stringify(cmds[name]))
```
Comment (tamuratak) [inline test/unittest.test.ts:71]: Please format the line to multiple lines for the readability.

**Review by tamuratak — CHANGES_REQUESTED** [review]

File: test/utils.ts (hunk 1 above)
Comment (jlelong) [reply test/utils.ts]: >  The word `check` is too general. It is not clear what is asserted.

I have turned this function into an assertion directly `assertDictKeyNames`. I hope this is clearer.

**Review by tamuratak — APPROVED** [review]

**Pull request merged**